src/Components/MainTableView/CheckDependency.tsx

```diff
@@ -0,0 +1,141 @@
+  constructor(props: any) {
+    super(props);
+    this.state = {
+      dependencies: [],
+      checkDependencyStatusMessage: '',
+      isGettingDependencies: false,
+    }
+  }
+
+  /**
+   * Checks whether parent component asked to clear the list of dependencies
+   * @param prevProps 
+   * @param prevState 
+   */
+  componentDidUpdate(prevProps: any, prevState: any) {
+    if (prevProps.clearList !== this.props.clearList && this.props.clearList) {
+      this.setState({dependencies: []})
+    }
+  }
+
+  /**
+   * Function to check table dependencies for the selected table entry
+   * @param entry
+   */
+  getDependencies(entry: any) {
+    // set status true for isGettingDependencies, switch to false once api responds
+    this.setState({isGettingDependencies: true})
+
+    // Check that tableAttirbutesInfo is not undefined
+    if (this.props.tableAttributesInfo === undefined) {
+      return;
+    }
+
+    // Copy the current state of processedEntry for processing for submission
+    let processedEntry = Object.assign({}, entry[0]?.primaryEntries);
+
+    // Loop through and deal with date, datetime, and timestamp formats
+    let tableAttributes: Array<TableAttribute> = this.props.tableAttributesInfo?.primaryAttributes as Array<TableAttribute>;
+    for (let tableAttribute of tableAttributes) {
+      if (tableAttribute.attributeType === TableAttributeType.DATE) {
+        // Convert date to DJ date format
+        processedEntry[tableAttribute.attributeName] = TableAttribute.parseDateToDJ(processedEntry[tableAttribute.attributeName])
+      }
+      else if (tableAttribute.attributeType === TableAttributeType.DATETIME || tableAttribute.attributeType === TableAttributeType.TIMESTAMP) {
+        // Convert to DJ friendly datetime format
+        processedEntry[tableAttribute.attributeName] = TableAttribute.parseDateTimeToDJ(processedEntry[tableAttribute.attributeName])
+      }
+    }
+
+    fetch(`/api/record/dependency?schemaName=${this.props.selectedSchemaName}&tableName=${this.props.selectedTableName}&restriction=${encodeURIComponent(btoa(JSON.stringify(processedEntry)))}`, 
+    {
+      method: 'GET',
+      headers: { 'Content-Type': 'application/json', 'Authorization': 'Bearer ' + this.props.token },
+      // body: JSON.stringify({schemaName: this.props.selectedSchemaName, tableName: this.props.selectedTableName, restriction: processedEntry})
+    })
+      .then(result => {
+        // Check for error mesage 500, if so throw error, but for now, send back dummy
+        if (result.status === 500) {
+          result.text().then(errorMessage => {
+            throw new Error(errorMessage)
+          }).catch(error => {
+            this.setState({checkDependencyStatusMessage: error.message, isGettingDependencies: false})
```

Comment:
Is the catch is needed when there is also another catch below?

Reply:
apparently yes - it kept on crashing saying unhandled promise until I put that catcher right next to it.

Reply:
Hmm I need to look into this closer later